### ISSUE TITLE
Sorting not working correctly

### ISSUE THREAD

**Issue by ThomasKranitsas** [appirio-tech/topcoder-app#1187]
Sorting works only on the visible challenges and not in all available challenges.
As a result when sorting challenges with `Prize high to low` I get this http://take.ms/d1p3s even if there are many challenges with prize > $0

fyi @ajefts

**Comment by ajefts** [comment]
@tladendo Can you get this fixes along with the v3 migration?

**Comment by tladendo** [comment]
@ajefts do you mean that sorting should apply to challenges that haven't been fetched from the API yet? I think we're going to just have to run a challenge to move more filtering to the backend...should i get that going?

**Comment by ajefts** [comment]
Yes.  For example, if there are 30 challenges open for registration and I sort for most recent first, then it should show the post recent out of the entire 30 challenges even if the UI only supports showing 10 total items...
